sqlglot/dialects/presto.py

```diff
@@ -294,6 +294,12 @@ class Generator(generator.Generator):
             ),
             exp.DateStrToDate: lambda self, e: f"CAST(DATE_PARSE({self.sql(e, 'this')}, {Presto.DATE_FORMAT}) AS DATE)",
             exp.DateToDi: lambda self, e: f"CAST(DATE_FORMAT({self.sql(e, 'this')}, {Presto.DATEINT_FORMAT}) AS INT)",
+            exp.DateSub: lambda self, e: self.func(
+                "DATE_ADD",
+                exp.Literal.string(e.text("unit") or "day"),
+                exp.Mul(this=e.expression, expression=exp.Literal.number(-1)),
```

Comment:
this causes a mutation on e.expression.parent let's do instead

e.expression * -1, which forces a copy and is cleaner